packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -794,6 +799,97 @@ Static group authorization should perform as expected.`,
     );
   };
 
+  addAutoGeneratedFields = (
+    ctx: TransformerContextProvider,
+    def: ObjectTypeDefinitionNode,
+    allowedFields: string[],
+    fields: readonly string[],
+  ) => {
+    this.addAutoGeneratedRelationalFields(ctx, def, allowedFields, fields);
+    this.addAutoGeneratedIndexFields(def, allowedFields);
+  };
+
+  addAutoGeneratedRelationalFields = (
+    ctx: TransformerContextProvider,
+    def: ObjectTypeDefinitionNode,
+    allowedFields: string[],
+    fields: readonly string[],
+  ) => {
+    this.addAutoGeneratedHasManyFields(ctx, def, allowedFields);
```

Comment:
we could do some pre-processing here before doing a level down collecting the typeDefs and re-mapping the field names.
ex. we can load all the types since its being called twice for hasmany and hasone.

ex.
```ts
const typeDefs = ctx.inputDocument.definitions.filter(it => it.kind === 'ObjectTypeDefinition')..map(it =>
        it.fields.map(field => {
          return { ...field, relatedType: it.name.value };
        }),
      );
```

Reply:
updated